ambry-commons/src/test/java/com.github.ambry.commons/HelixNotifierTest.java

```diff
@@ -0,0 +1,365 @@
+  private HelixNotifier notifier;
+  private List<TopicListener<String>> listeners;
+
+  @BeforeClass
+  public static void initialize() throws Exception {
+    storeConfig = HelixPropertyStoreUtils.getHelixStoreConfig(zkClientConnectString, zkClientSessionTimeoutMs,
+        zkClientConnectTimeoutMs, storeRootPath);
+  }
+
+  @Before
+  public void setup() throws Exception {
+    resetTopicListeners();
+    HelixPropertyStoreUtils.deleteStoreIfExists(storeConfig, storeFactory);
+    listeners = new ArrayList<>();
+    notifier = new HelixNotifier(storeConfig, storeFactory);
+    listeners.add(new ListenerForTest(latch0, receivedTopicsByListener0, receivedMessagesByListener0));
+    listeners.add(new ListenerForTest(latch1, receivedTopicsByListener1, receivedMessagesByListener1));
+  }
+
+  @AfterClass
+  public static void cleanStore() throws Exception {
+    HelixPropertyStoreUtils.deleteStoreIfExists(storeConfig, storeFactory);
+  }
+
+  /**
+   * Tests normal operations using a single {@link HelixNotifier} and two {@link TopicListener}s.
+   * @throws Exception Any unexpected exceptions.
+   */
+  @Test
+  public void testHelixNotifier() throws Exception {
+    // Subscribe a topic and send a message for the topic
+    notifier.subscribe(refTopics.get(0), listeners.get(0));
+    latch0.set(new CountDownLatch(1));
+    notifier.sendMessage(refTopics.get(0), refMessages.get(0));
+    awaitLatchOrTimeout(latch0.get(), latchTimeoutMs);
+    assertEquals("Received topic is different from expected", refTopics.get(0), receivedTopicsByListener0.get(0));
+    assertEquals("Received message is different from expected", refMessages.get(0), receivedMessagesByListener0.get(0));
+
+    // Send a different message for the topic
+    resetTopicListeners();
+    latch0.set(new CountDownLatch(1));
+    notifier.sendMessage(refTopics.get(0), refMessages.get(1));
+    awaitLatchOrTimeout(latch0.get(), latchTimeoutMs);
+    assertEquals("Received topic is different from expected", refTopics.get(0), receivedTopicsByListener0.get(0));
+    assertEquals("Received message is different from expected", refMessages.get(1), receivedMessagesByListener0.get(0));
+
+    // Subscribe to a different topic and send message for that topic
+    resetTopicListeners();
+    notifier.subscribe(refTopics.get(1), listeners.get(0));
+    latch0.set(new CountDownLatch(1));
+    notifier.sendMessage(refTopics.get(1), refMessages.get(0));
+    awaitLatchOrTimeout(latch0.get(), latchTimeoutMs);
+    assertEquals("Received topic is different from expected", refTopics.get(1), receivedTopicsByListener0.get(0));
+    assertEquals("Received message is different from expected", refMessages.get(0), receivedMessagesByListener0.get(0));
+  }
+
+  /**
+   * Tests non-alphabetNumeric characters.
+   * @throws Exception Any unexpected exceptions.
+   */
+  @Test
+  public void testNonAlphabetNumericMessage() throws Exception {
+    String nonAlphabetNumericMessage = "!@#$%^&*(){}|?><~";
```

Comment:
This is to ensure that it is ok to send any message to a topic, though I agree with you that this should be taken care of by Helix or ZK unit tests. I can remove the test if you feel so.